core/option.go

```diff
@@ -17,7 +18,7 @@ type params struct {
 
 func defaultParams() params {
 	return params{
-		availabilityWindow: time.Duration(0),
+		availabilityWindow: availability.StorageWindow,
```

Comment:
this is the most important change - we were not passing the availability window properly to the listener + to the core fetcher, making this bug possible.